upup/pkg/fi/cloudup/scalewaytasks/lb_backend.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scalewaytasks
+
+import (
+	"fmt"
+
+	"github.com/scaleway/scaleway-sdk-go/api/lb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/scaleway"
+)
+
+type LBBackend struct {
+	Name      *string
+	Lifecycle fi.Lifecycle
+
+	ID                   *string
+	LBName               *string
+	Zone                 *string
+	ForwardProtocol      *string
+	ForwardPort          *int32
+	ForwardPortAlgorithm *string
+	StickySessions       *string
+	ProxyProtocol        *string
+}
+
+var _ fi.CloudupTask = &LBBackend{}
+var _ fi.CompareWithID = &LBBackend{}
+var _ fi.HasName = &LBBackend{}
+
```

Comment:
This should either have a pointer to the `LoadBalancer` it depends on or it should implement `HasDependencies` and return the `LoadBalancer` from `GetDependencies`. The former is easier. That will cause the `Task` scheduler to run the `LoadBalancer` task to completion before starting the `LBBackend`.

You can then get the `ID` out of the `LoadBalancer` struct, saving the need to list the LBs again.

Reply:
Thanks for the explanation, I didn't know about these options :+1: